nscplugin/src/main/scala/scala/scalanative/nscplugin/NirCodeGen.scala

```diff
@@ -1540,12 +1540,33 @@ abstract class NirCodeGen
           val offset = genExpr(offsetp, ptr)
           offset withOp Op.Elem(ty, ptr.value, Seq(offset.value))
 
-        case (PTR_SUB, Seq(offsetp, tagp)) =>
-          val st     = unwrapTag(tagp)
-          val ty     = genType(st, box = false)
-          val offset = genExpr(offsetp, ptr)
-          val neg    = negateInt(offset.value, offset)
-          neg withOp Op.Elem(ty, ptr.value, Seq(neg.value))
+        case (PTR_SUB, Seq(argp, tagp)) =>
+          argp match {
+            case lit: Literal =>
+              val st     = unwrapTag(tagp)
+              val ty     = genType(st, box = false)
+              val offset = genExpr(argp, ptr)
+              val neg    = negateInt(offset.value, offset)
+              neg withOp Op.Elem(ty, ptr.value, Seq(neg.value))
+
+            case id: Ident =>
+              def show(f: Focus): Focus = {
+                f.insts.foreach(i => println("  |" + i))
+                f
+              }
+              /*
+
+              With tag type
+              val st     = unwrapTag(tagp)
+              val ty     = genType(st, box = false)
+              val ptrInt    = ptrArg withOp Op.Conv(nir.Conv.Ptrtoint, ty, ptr.value)
+              val ptrArgInt = ptrInt withOp Op.Conv(nir.Conv.Ptrtoint, ty, ptrArg.value)
+              */
+              val ptrArg    = genExpr(argp, ptr)
+              val ptrInt    = ptrArg withOp Op.Conv(nir.Conv.Ptrtoint, nir.Type.ULong, ptr.value)
+              val ptrArgInt = ptrInt withOp Op.Conv(nir.Conv.Ptrtoint, nir.Type.ULong, ptrArg.value)
```

Comment:
Found the reason: https://github.com/scala-native/scala-native/blob/master/tools/src/main/scala/scala/scalanative/codegen/CodeGen.scala#L329

Given that
> Corresponds to LLVM’s pointer type with a major distinction of not preserving the type of memory that’s being pointed at. Pointers are going to become untyped in LLVM in near future too.

the type will have to be handled explicitly by the generated code.